src/app/auth/profiles/profiles.component.ts

```diff
@@ -91,12 +91,24 @@ export class ProfilesComponent implements OnInit {
   }
 
   /**
-   * Processes available Profiles information from application's configuration.
+   * Gets the required profiles from the external repository file.
+   */
+   private fetchExternalProfiles(): Promise<Profile[]> {
```

Comment:
If a user selects a `profiles.json` file it erases the data that is fetched from the `client_data` repo since the `AppConfig.profiles` is concatenated to the profiles read from the `json` 😅 I think the expected behaviour should be that the new profile is concatenated to those that are in `client_data` Maybe can either just re-fetch or  concatenate to existing profiles instead of overriding the `this.profiles` variable.

https://github.com/CATcher-org/CATcher/blob/b8f420a02809dc8c912a4fd729cda00ebe554d83/src/app/auth/profiles/profiles.component.ts#L71-L91

Reply:
Ah yes, this sounds like a better alternative to deal with custom profiles! 👍 

I'll fix that shortly

Reply:
Should have been fixed! Do let me know if I misunderstood what you have said ><